### ISSUE TITLE
[Question]: LlamaIndex retriever on Confluence pages

### ISSUE THREAD

**Issue by hz6yc3** [run-llama/llama_index#16958]
### Question Validation

- [X] I have searched both the documentation and discord for an answer.

### Question

I am using the LlamaIndex provided Confluence Reader to ingest confluence pages. I am using the SemanticSplitterNodeParser to parse the confluence pages and generate nodes and indexing them in the vector store which works fine. I have created a context chat engine using the index as retriever (tried the simple as_retriever(), RecursiveRetriever(), etc.)

The confluence pages are structured in such a way that there is a unique ticket number in the page title (page heading) and the contents of the page provide description about the ticket, when the ticket was created and other details about it. The page content has no mention of the ticket number and is present only on the page title. The problem I am facing is that when the user is asking a question with reference to the ticket number (like - Can you summarize the ticket XYZ?) which is present only on the page title, the retrievers are consistently behaving poorly and do not retrieve the related document chunks along the page title is present in the node metadata/document metadata. I tried various methods like using DocumentSummaryIndex to summarize the page and creating a retriever on top of it but none of them seem to be working as expected in this case. I even tried to copy the page title into the document text but still no luck.

The retrievers are behaving very weird and not pulling the context correctly. No other post processing technique like using re-rankers, etc. would help because the retrieval itself is bad and doesn't pull the context. Any suggestions on this issue?

@dosu

**Comment by hz6yc3** [comment]
Thanks! I have dates embedded within the context or the page content. How can I use a filter condition on the date to pull only relevant chunks @dosu

**Comment by hz6yc3** [comment]
@dosu

**Comment by hz6yc3** [comment]
@dosu
This is not working. I am not able to find the related libraries for extracting structured data from text

**Comment by logan-markewich** [comment]
@hz6yc3 semantic retrieval performs pretty badly for the scenario you suggested yes.

I would recommend using BM25 as your retriever (or some kind of hybrid retrieval offered by whichever vector store you are using), or extracting out information like that into the metadata so that you can use an llm to infer metadata filters

**Comment by hz6yc3** [comment]
@logan-markewich thanks for your response. Yes BM25 is definitely performing better but for extracting the embedded dates in the text like you suggested I was looking for something out of the box and stumbled on Structured Data Extraction using Llama Extract which will help extract the date and add it to the node metadata but unfortunately Llama Extract requires a API key and I am not sure if I can use it within our company. May be I will try using another NLP tool just to extract the date like Spacy or something
